core/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportClusterAllocationExplainAction.java

```diff
@@ -139,14 +213,35 @@ public static ClusterAllocationExplanation explainShard(ShardRouting shard, Rout
                 nodeToDecision.put(discoNode, d);
             }
         }
-        long remainingDelayNanos = 0;
+        long remainingDelayMillis = 0;
+        final MetaData metadata = allocation.metaData();
+        final IndexMetaData indexMetaData = metadata.index(shard.index());
         if (ui != null) {
-            final MetaData metadata = allocation.metaData();
-            final Settings indexSettings = metadata.index(shard.index()).getSettings();
-            remainingDelayNanos = ui.getRemainingDelay(System.nanoTime(), metadata.settings(), indexSettings);
+            final Settings indexSettings = indexMetaData.getSettings();
+            long remainingDelayNanos = ui.getRemainingDelay(System.nanoTime(), metadata.settings(), indexSettings);
```

Comment:
Looking at this again, I think we can remove the node settings as updateDelay / getRemainingDelay only depends on index settings.